common/raw_hashtable.h

```diff
@@ -0,0 +1,1206 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_COMMON_RAW_HASHTABLE_H_
+#define CARBON_COMMON_RAW_HASHTABLE_H_
+
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <new>
+#include <type_traits>
+#include <utility>
+
+#include "common/check.h"
+#include "common/hashing.h"
+#include "common/raw_hashtable_metadata_group.h"
+#include "llvm/ADT/Sequence.h"
+#include "llvm/Support/Compiler.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/MathExtras.h"
+
+// A namespace collecting a set of low-level utilities for building hashtable
+// data structures. These should only be used as implementation details of
+// higher-level data structure APIs.
+//
+// These utilities support hashtables following a *specific* API design pattern,
+// and using SSO or Small-Size Optimization when desired. We expect there to be
```

Comment:
The text currently reads like SSO and Small-Size Optimization are two distinct alternatives.
```suggestion
// and using Small-Size Optimization (or "SSO") when desired. We expect there to be
```

or

```suggestion
// and using Small-Size Optimization, or "SSO", when desired. We expect there to be
```